app/ts-cli/geminicli/export.go

```diff
@@ -0,0 +1,1786 @@
+		database:    "",
+		measurement: "",
+		startTime:   0,
+		endTime:     0,
+	}
+}
+
+func (d *dataFilter) parseTime(clc *CommandLineConfig) error {
+	var start, end string
+	timeSlot := strings.Split(clc.TimeFilter, "~")
+	if len(timeSlot) == 2 {
+		start = timeSlot[0]
+		end = timeSlot[1]
+	} else if clc.TimeFilter != "" {
+		return fmt.Errorf("invalid time filter %q", clc.TimeFilter)
+	}
+	// set defaults
+	if start != "" {
+		st, err := convertTime(start)
+		if err != nil {
+			return err
+		}
+		d.startTime = st
+	} else {
+		d.startTime = math.MinInt64
+	}
+
+	if end != "" {
+		ed, err := convertTime(end)
+		if err != nil {
+			return err
+		}
+		d.endTime = ed
+	} else {
+		// set end time to max if it is not set.
+		d.endTime = math.MaxInt64
+	}
+
+	if d.startTime > d.endTime {
+		return fmt.Errorf("start time `%q` > end time `%q`", start, end)
+	}
+
+	return nil
+}
+
+func (d *dataFilter) parseDatabase(dbFilter string) {
+	if dbFilter == "" {
+		return
+	}
+	d.database = dbFilter
+}
+
+func (d *dataFilter) parseRetention(retentionFilter string) {
+	if retentionFilter == "" {
+		return
+	}
+	d.retention = retentionFilter
+}
+
+func (d *dataFilter) parseMeasurement(mstFilter string) error {
+	if mstFilter == "" {
+		return nil
+	}
+	if mstFilter != "" && d.database == "" {
+		return fmt.Errorf("measurement filter %q requires database filter", mstFilter)
+	}
+	d.measurement = mstFilter
+	return nil
+}
+
+// timeFilter [startTime, endTime]
+func (d *dataFilter) timeFilter(t int64) bool {
+	return t >= d.startTime && t <= d.endTime
+}
+
+func (d *dataFilter) isBelowMinTimeFilter(t int64) bool {
+	return t < d.startTime
+}
+
+func (d *dataFilter) isAboveMaxTimeFilter(t int64) bool {
+	return t > d.endTime
+}
+
+type DatabaseDiskInfo struct {
+	dbName          string              // ie. "NOAA_water_database"
+	rps             map[string]struct{} // ie. ["0:autogen","1:every_one_day"]
+	dataDir         string              // ie. "/tmp/openGemini/data/data/NOAA_water_database"
+	walDir          string              // ie. "/tmp/openGemini/data/wal/NOAA_water_database"
+	rpToTsspDirMap  map[string]string   // ie. {"0:autogen", "/tmp/openGemini/data/data/NOAA_water_database/0/autogen"}
+	rpToWalDirMap   map[string]string   // ie. {"0:autogen", "/tmp/openGemini/data/wal/NOAA_water_database/0/autogen"}
+	rpToIndexDirMap map[string]string   // ie. {"0:autogen", "/tmp/openGemini/data/data/NOAA_water_database/0/autogen/index"}
+}
+
+func newDatabaseDiskInfo() *DatabaseDiskInfo {
+	return &DatabaseDiskInfo{
+		rps:             make(map[string]struct{}),
+		rpToTsspDirMap:  make(map[string]string),
+		rpToWalDirMap:   make(map[string]string),
+		rpToIndexDirMap: make(map[string]string),
+	}
+}
+
+func (d *DatabaseDiskInfo) init(actualDataDir string, actualWalDir string, databaseName string, retentionPolicy string) error {
+	d.dbName = databaseName
+
+	// check whether the database is in actualDataPath
+	dataDir := path.Join(actualDataDir, databaseName)
```

Comment:
`path.Join` use unix style slash `/` to combie string, on windows platform may cause problem.